rust/src/main.rs

```diff
@@ -1,135 +1,230 @@
-use futures_lite::future::FutureExt;
-use std::env::{current_dir, vars};
-use std::sync::Arc;
-use tokio::signal;
+//! Echo bot example in Rust.
+//!
+//! This is a bot example which echo's back the text of any message it receives in direct
+//! 1:1 chats.  It contains a number of steps:
+//!
+//! - Create the directory for deltachat storage.  Deltachat is statefull and all the state
+//!   is stored here.
+//! - Creates the [`Context`].
+//! - On the first run: configures the credentials.
+//! - Spawns a Tokio task to receive and handle events from deltachat-core.
+//! - Starts the tasks for deltchat-core to send and receive messages.
+//! - Waits for the shutdown signal.
+//! - Stops the tasks for deltachat-core to send and receive messages.
+//!
+//! Central to deltachat-core is the [`Context`], it is a struct which contains all the
+//! state to run a single DeltaChat account.  Once a [`Context`] is created all operations
+//! work on this context.
+//!
+//! # Running the bot
+//!
+//! To run the bot, the first time you need to set the credentials in the environment:
+//!
+//! ```shell
+//! addr=$yourEmail mail_pw=$yourPassword cargo run
+//! ```
+//!
+//! This will create a subdirectory called `deltachat-db` in the current working directory,
+//! this is where deltachat stores the state.  The credentials will be stored here so
+//! further invocations do not need them specified again if the `deltachat-db` directory can
+//! be found again.
+//!
+//! # API Stability
+//!
+//! It should be noted there is currently no API stability guarantee for the Rust API.  The
+//! only API stability provided by the DeltaChat library is on the C API via the
+//! `deltachat_ffi` crate.  However when using Rust it is probably better to live with the
+//! API instability instead of going via the unsafe `deltachat_ffi` crate.  In practice the
+//! API does not change drastically between tagged releases.
```

Comment:
(and they can pin the core version anyway, so yes better to use Rust API directly instead of CFFI)